homeassistant/components/fritz/__init__.py

```diff
@@ -53,13 +53,34 @@ def _async_unload(event):
     entry.async_on_unload(
         hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _async_unload)
     )
+
+    async_device_setup(hass, entry, fritz_tools)
+
     # Load the other platforms like switch
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
 
     return True
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: ConfigType) -> bool:
+@callback
+def async_device_setup(
+    hass: HomeAssistant, entry: ConfigEntry, fritz_tools: FritzBoxTools
+):
+    """Set up a device that is online."""
+    dev_reg = device_registry.async_get(hass)
+    dev_reg.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        name=entry.title,
+        connections={(device_registry.CONNECTION_NETWORK_MAC, fritz_tools.mac)},
+        # This is duplicate but otherwise via_device can't work
+        identifiers={(DOMAIN, fritz_tools.mac)},
+        manufacturer="AVM",
+        model=fritz_tools.model,
+        sw_version=fritz_tools.sw_version,
+    )
+
```

Comment:
Since the fritz box itself will have an entity now you don't need to create it manually and can  set `device_info` on the new binary sensor's base class

Reply:
Done